compiler/rustc_const_eval/src/const_eval/error.rs

```diff
@@ -138,9 +138,10 @@ where
             ErrorHandled::TooGeneric(span.unwrap_or(DUMMY_SP))
         }
         err_inval!(AlreadyReported(guar)) => ErrorHandled::Reported(guar, span.unwrap_or(DUMMY_SP)),
-        err_inval!(Layout(LayoutError::ReferencesError(guar))) => {
-            ErrorHandled::Reported(guar.into(), span.unwrap_or(DUMMY_SP))
-        }
+        err_inval!(Layout(LayoutError::ReferencesError(guar))) => ErrorHandled::Reported(
```

Comment:
What if there is a`SizeOverflow` in a promoted?

This PR seems like a good step so I am okay with landing it, but it seems to me there's still a potential for ICEs here.

Reply:
I added a test for promoteds. On master we got a bogus "erroneous constant" *note*. Now it's gone.

Reply:
I was specifically wondering about `SizeOverflow`  as that's a different `LayoutError` variant that would bypass your new case here,